### PR TITLE
Username exists

### LINKED ISSUE #9333: "Username Exists" for facility change UI is implemented
This is a final issue to ensure that after the state machine, routes, and pages have been created, that the "Username Exists" flow works as expected. 

Any code adjustments necessary to ensure that the feature is completed is considered in scope for this issue.

### PR DESCRIPTION
**Pull request by jredrejo**
## Summary
Creation of the screen that warns the user an existing username exists in the target facility before changing.

image

This PR has also been used to refactor the  `isIndividual` property/url to `onMyOwnSetup`

## References
Closes: #9333
Depends: #9577
- [Figma designs](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=510%3A5829)

## Reviewer guidance
Setup:
- Sets the on_my_own_setup facility option as explained in https://github.com/learningequality/kolibri/pull/9577
- Sets two kolibri servers as in https://github.com/learningequality/kolibri/pull/9576
- Create users with the same username both in the source and target facility before beginning the change facility flow.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by MisRob** [comment]
@jredrejo Is this testable already or will it be possible rather after we merge https://github.com/learningequality/kolibri/pull/9577 and rebase this PR?

**Comment by MisRob** [comment]
Ah I see now, it has the same changes here so I think it should be possible

**Comment by jredrejo** [comment]
> @jredrejo Is this testable already or will it be possible rather after we merge #9577 and rebase this PR?

yes, and I have also just rebased develop with the latest changes

File: kolibri/plugins/user_profile/assets/test/views/ChangeFacility/UsernameExists.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import { mount, createLocalVue } from '@vue/test-utils';
+import VueRouter from 'vue-router';
+import makeStore from '../../makeStore';
+import UsernameExists from '../../../src/views/ChangeFacility/UsernameExists';
+
+const localVue = createLocalVue();
+localVue.use(VueRouter);
```
Comment (MisRob) [inline kolibri/plugins/user_profile/assets/test/views/ChangeFacility/UsernameExists.spec.js]: I don't see any router dependencies in the component being tested so I think tests should work without the need to import  and use VueRouter

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      function to_merge() {
+        this.changeFacilityService.send({
+          type: 'MERGE',
+        });
+      }
+
+      return { firstLine, secondLine, to_create, to_merge };
+    },
+
+    inject: ['changeFacilityService', 'state'],
```
Comment (rtibbles) [inline kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue]: Can set this up using the composition API too - can import `inject` from `'kolibri.lib.vueCompositionApi'`, and then assign it to a variable inside `setup`:
```
const changeFacilityService = inject('changeFacilityService');
```

No need to reference this inside the methods you are creating in that case.

Comment (jredrejo) [reply kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue]: Good advice, thanks. Done!

**Review by rtibbles — APPROVED** [review]

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    components: { BottomAppBar },
+
+    mixins: [commonCoreStrings],
+    setup() {
+      const changeFacilityService = inject('changeFacilityService');
+      const state = inject('state');
+
+      const firstLine = computed({
+        get() {
+          return this.$tr('changeFacilityInfoLine1', {
```
Comment (MisRob) [inline kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue:55]: Oh nice, you find a way of accessing `$tr` in `setup`. For some reason, I wasn't able to use it when I first experimented with Composition API. Was it defining the computed getter that helped or something else?

**Review by MisRob — APPROVED** [review]
Thank you, @jredrejo, code changes are looking good to me, and I can confirm that it works on my machine. Left one minor comment, nothing blocking.

File: kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue (hunk 3 above)
Comment (jredrejo) [reply kolibri/plugins/user_profile/assets/src/views/ChangeFacility/UsernameExists.vue:55]: yep, using getters/setters as  explained here https://vuejs.org/guide/essentials/computed.html#writable-computed did the magic. If the getter is not used it seems `this` is not bound correctly

File: kolibri/plugins/user_profile/assets/test/views/ChangeFacility/UsernameExists.spec.js (hunk 1 above)
Comment (jredrejo) [reply kolibri/plugins/user_profile/assets/test/views/ChangeFacility/UsernameExists.spec.js]: yeah, good catch, I have removed the store too, as it's not used either

**Pull request merged**